framework/tlvf/src/src/WSC/m1.cpp

```diff
@@ -183,8 +223,84 @@ bool m1::init(const config &cfg)
     return true;
 }
 
-bool m1::valid_custom() const
+bool m1::valid() const
```

Comment:
Is this validation really worthwhile? Do we really need to fail if an agent/controller doesn't provide a version attribute?

For the attributes that we really use, there is already implicit validation in the agent/controller code that handles the M2/M1.

Oh, hang on, we don't actually validate those things in the handler code because we call wrapper functions that dereference the attributes. However, aren't those wrappers going to trigger klockwork errors because they dereference a potentially nullptr? If they do, I think we need to add an assert in those wrapper functions. Actually, I think we need to add asserts in many places in the code :-) To be discussed...

Reply:
The validation is exactly for this - since our wrappers are accessing attributes directly.
KW is not an issue - we will mark these as non-issues.

Reply:
I guess we can be less strict if we see that some controller don't send some attributes, but for now I think its OK to be strict - we are just starting to test with other controllers so I guess we'll find out.